ban/commands/municipality.py

```diff
@@ -0,0 +1,110 @@
+from ban.commands import command, reporter
+from ban.core import models
+from ban.core import versioning
+from . import helpers
+
+
+@command
+@helpers.session
+def merge(destination, sources=[], name='', label='', **kwargs):
+    """
+    Municipality merge command.
+    Steps:
+    - for each Municipality to be removed:
+    - create a new Group with its name
+    - attach all HouseNumbers of this Municipality to this new Group
+    - attach sources Groups and sources PostCodes to destination
+    - Remove the Municipality
+    """
+    if destination in sources:
+        helpers.abort('Destination in sources')
+    if not name or not name.split():
+        helpers.abort('Name should not be empty')
+    if not label or not label.split():
+        helpers.abort('Label should not be empty')
+    try:
+        destination = models.Municipality.get(
+            models.Municipality.insee == destination)
+    except models.Municipality.DoesNotExist:
+        helpers.abort('Destination does not exist')
+    if not sources:
+        helpers.abort('No sources')
+    sources_inst = []
+    # Make sure all sources exist before processing any of them.
+    for source in sources:
+        try:
+            source = models.Municipality.get(
+                models.Municipality.insee == source)
+        except models.Municipality.DoesNotExist:
+            helpers.abort('Source {} does not exist'.format(source))
+        else:
+            sources_inst.append(source)
+    source_done = []
+    areas = []
+    db = models.Municipality._meta.database
+    db.begin()
+    for source in sources_inst:
+        if source.insee not in source_done:
+            source_done.append(source.insee)
+            process_source(destination, source, areas, label)
+    process_destination(destination, areas, name, label)
+    print(reporter)
+    if helpers.confirm('Do you feel confident with those changes ?'):
+        db.commit()
+    else:
+        db.rollback()
+
+
+def process_source(destination, source, areas, label):
+    versioning.Redirect.add(destination, 'insee', source.insee)
+    versioning.Redirect.add(destination, 'id', source.id)
+    reporter.notice('redirected to destination', source)
+    process_postcode(destination, source, label)
+    process_source_to_group(destination, source, areas, label)
+    source.delete_instance()
+
+
+def process_destination(destination, areas, name, label):
+    process_postcode(destination, destination, label)
+    process_source_to_group(destination, destination, areas, label)
+    destination.name = name
+    destination.increment_version()
+    destination.save()
+    reporter.notice('name modified', destination)
+
+
+def process_source_to_group(destination, source, areas, label):
```

Comment:
Given it's not only used by the source but also by the destination, I'd name it differently.
Maybe "group_from_municipality", no need for "process_" in this case anyway.